### PR TITLE
Added CMK schema, introduced type import & grouped interface files

### PR DESCRIPTION
**Pull request by AlexanderSehr**
# Overview/Summary

Redone version of #1737

- Added CMK schema as per https://github.com/Azure/bicep-registry-modules/issues/2842#issuecomment-2423679879
- Replaced copied schemas with avm-common-types import
- Moved interface files for tf into a tf folder, and bicep files into a bicep folder for easier maintainability

image

### CODE REVIEW

File: docs/content/specs-defs/specs/terraform/resource/interfaces.md [hunk 1]
```diff
@@ -125,25 +125,11 @@ In Terraform, locks become part of the resource graph and suitable `depends_on`
...
-  {{< expand "➕ Variant 1: A default service (`groupId`) can be assumed - e.g., for services that only have one private endpoint type)" "expand/collapse" >}}
-  {{< include file="/static/includes/interfaces/int.pe.udt.schema1.bicep" language="bicep" options="linenos=false" >}}
-  {{< /expand >}}
-  {{< expand "➕ Variant 2: A default service (`groupId`) cannot be assumed - e.g., for services that have more than one private endpoint type, like a Storage Account (blob, file, etc.)" "expand/collapse" >}}
-  {{< include file="/static/includes/interfaces/int.pe.udt.schema2.bicep" language="bicep" options="linenos=false" >}}
-  {{< /expand >}}
-  {{< /tab >}}
-  {{< tab "Bicep Input Example with Values" >}}
-  {{< include file="/static/includes/interfaces/int.pe.input.bicep" language="bicep" options="linenos=false" >}}
-  {{< /tab >}}
```
Comment (eriqua) [inline docs/content/specs-defs/specs/terraform/resource/interfaces.md:141]: Differently than others just moved, this content looks removed. Just double checking if that's intended

File: docs/static/includes/interfaces/bicep/int.locks.udt.schema.bicep [hunk 2]
```diff
@@ -3,8 +3,9 @@
 //   Parameters   //
 // ============== //
 
+import { lockType } from 'br/public:avm/utl/types/avm-common-types:0.4.0'
```
Comment (eriqua) [inline docs/static/includes/interfaces/bicep/int.locks.udt.schema.bicep:6]: How do we want to maintain this versioning? Are we going to update all schema files every time we update the common type module? Not to necessarily address in this PR, but something to be aligned upon

File: docs/static/includes/interfaces/bicep/int.diag.udt.schema.bicep [hunk 3]
```diff
@@ -0,0 +1,35 @@
+
+// ============== //
+//   Parameters   //
+// ============== //
+
+import { diagnosticSettingFullType } from 'br/public:avm/utl/types/avm-common-types:0.4.0'
```
Comment (eriqua) [inline docs/static/includes/interfaces/bicep/int.diag.udt.schema.bicep:6]: Should we document the other 2 diagnostic setting variants as well?

File: docs/static/includes/interfaces/tf/int.cmk.input.tf
Comment (eriqua) [inline docs/static/includes/interfaces/tf/int.cmk.input.tf:1]: I'd suggest to double check with the TF team if updating these paths will have any impact on the their testing framework. If not, the folder restructure looks good to me

File: docs/static/includes/interfaces/bicep/int.locks.udt.schema.bicep (hunk 2 above)
Comment (AlexanderSehr) [reply docs/static/includes/interfaces/bicep/int.locks.udt.schema.bicep:6]: Fair point. We can also replace it with `<version>` as we are doing in the readmes

Comment (AlexanderSehr) [reply docs/static/includes/interfaces/bicep/int.locks.udt.schema.bicep]: Addressed

File: docs/static/includes/interfaces/tf/int.cmk.input.tf
Comment (AlexanderSehr) [reply docs/static/includes/interfaces/tf/int.cmk.input.tf:1]: We can - we did with PE afterall. Should we then do this for all variants of all UDTs? Also also for e.g. Managed Identities?

File: docs/static/includes/interfaces/bicep/int.diag.udt.schema.bicep (hunk 3 above)
Comment (AlexanderSehr) [reply docs/static/includes/interfaces/bicep/int.diag.udt.schema.bicep]: We can - we did with PE afterall. Should we then do this for all variants of all UDTs? Also also for e.g. Managed Identities?

File: docs/content/specs-defs/specs/terraform/resource/interfaces.md (hunk 1 above)
Comment (AlexanderSehr) [reply docs/content/specs-defs/specs/terraform/resource/interfaces.md:141]: That was intentional and a mistake by the original auther. If you look closely, it references Bicep code, even though it is the TF. The correct TF code is referenced below

File: docs/static/includes/interfaces/tf/int.cmk.input.tf
Comment (AlexanderSehr) [reply docs/static/includes/interfaces/tf/int.cmk.input.tf:1]: Just got the confirmation by @matt-FFFFFF that we're good to go :)

File: docs/static/includes/interfaces/bicep/int.diag.udt.schema.bicep (hunk 3 above)
Comment (AlexanderSehr) [reply docs/static/includes/interfaces/bicep/int.diag.udt.schema.bicep]: Added examples & tabs for **all** variants 

image

**Review by eriqua — APPROVED** [review]
🆗

**Pull request merged**